cmplibrary/src/main/java/com/sourcepoint/cmplibrary/campaign/CampaignManager.kt

```diff
@@ -406,25 +406,32 @@ private class CampaignManagerImpl(
 
     override val shouldCallConsentStatus: Boolean
         get() {
-            val gdprUUID = dataStorage.gdprConsentUuid
-            val ccpaUUID = dataStorage.ccpaConsentUuid
-            val localStateSize = messagesOptimizedLocalState?.jsonObject?.size ?: 0
+            val isGdprOrCcpaUuidPresent =
+                dataStorage.gdprConsentUuid != null || dataStorage.ccpaConsentUuid != null
+            val isLocalStateEmpty = messagesOptimizedLocalState?.jsonObject?.isEmpty() == true
             val isV6LocalStatePresent = dataStorage.preference.all.containsKey(LOCAL_STATE)
-            val isV6LocalStatePresent2 = dataStorage.preference.all.containsKey(DataStorage.LOCAL_STATE_OLD)
-            val res =
-                ((gdprUUID != null || ccpaUUID != null) && localStateSize == 0) || isV6LocalStatePresent || isV6LocalStatePresent2
+            val isV6LocalStatePresent2 = dataStorage.preference.all.containsKey(LOCAL_STATE_OLD)
+            val hasNonEligibleLocalDataVersion =
+                dataStorage.localDataVersion != DataStorage.LOCAL_DATA_VERSION_HARDCODED_VALUE
+
+            val isRequireNewConsentStatusData = (isGdprOrCcpaUuidPresent && isLocalStateEmpty) ||
+                isV6LocalStatePresent ||
+                isV6LocalStatePresent2 ||
+                hasNonEligibleLocalDataVersion
 
             logger?.computation(
```

Comment:
@andresilveirah I strongly believe that we need `logger` for future debugging, it is easier to debug when you have all the logs in place before you start the investigation.

So count this up as an investment for the future.

Reply:
The loggers are getting on the way of readability. Right now, we're so dependent on them because we debug the SDK a lot. And we debug the SDK a lot because we have so many bugs, but we're changing that. I rather invest in simplifying the code and having it more readable than super instrumented and hard to read.